### ISSUE TITLE
Allow compiling PSA without Cipher when built-in ciphers/AEADs/CMAC are not needed.

### ISSUE THREAD

**Issue by mpg** [Mbed-TLS/mbedtls#6317]
Currently `MBEDTLS_PSA_CRYPTO_C` has a hard dependency on `MBEDTLS_CIPHER_C`. However, is should be possible to build PSA Crypto without Cipher when we the built-in implementations of ciphers & AEADs are not needed, which happens in two cases:

1. no cipher or AEAD (edit: other than ChachaPoly) (edit: or CMAC) is requested, that is all the corresponding `PSA_WANT_xxx` symbols are undefined;
2. for each cipher or AEAD (edit: other than ChachaPoly) (edit: or CMAC) that is requested, a driver is provided, so that the built-in implementation is not needed.

This task is to make those two cases work. The second scenario will be tested by disabling `MBEDTLS_CIPHER_C` in the component created in #6316. The first scenario will be tested by adding a new component.

It might be useful to create an internal `cipher_info` module similar to `hash_info`.

Depends on: #6316.

**Comment by gilles-peskine-arm** [comment]
It should also be possible to build PSA without cipher.c when the only cipher/AEAD algorithm in use is chachapoly, because the AEAD built-in code doesn't dispatch through cipher.

This isn't possible with CCM or GCM because the lower-level modules use the cipher module themselves. It isn't currently possible for non-auth ciphers because psa_crypto_cipher.c is built on top of cipher.c, though we'd like to change this at some point to remove the intermediate layer, or make PSA dispatch the primary and cipher a compatibility wrapper.

**Comment by mpg** [comment]
Good point, thanks! I edited the description.

**Comment by gilles-peskine-arm** [comment]
I missed this issue when I filed https://github.com/Mbed-TLS/mbedtls/issues/8048. https://github.com/Mbed-TLS/mbedtls/issues/8048 is mostly a duplicate of this, but not exactly: it doesn't consider driver-only builds. #8048 is basically part 1 of #6317. So I'm keeping both issues open.

**Comment by mpg** [comment]
This issue's part 1 is already covered by another issue, and part 2 was based on an outdated strategy for driver-only cipher support, and is unlikely to be be relevant with the new strategy, hence closing.
